views/house_wikidata.erb

```diff
@@ -18,3 +19,34 @@
       </ol>
   </div>
 </div>
+
+<script type="text/javascript">
+  $(document).ready(function() {
+    var wikidata_id = '<%= @page.wikidata_id %>',
+        endpointUrl = 'https://query.wikidata.org/sparql',
+        sparqlQuery = "SELECT ?seat_count\n" +
+                      "WHERE { wd:" + wikidata_id + " wdt:P1342 ?seat_count }",
+        settings = {
+          headers: { Accept: 'application/sparql-results+json' },
+          data: { query: sparqlQuery }
+        };
+
+    $.ajax(endpointUrl, settings).done(function(data) {
+      if (data.results.bindings.length == 1) {
```

Comment:
I know there are different views on this, but I prefer the Crockford style of Javascript where you explicitly hoist the variable declarations to the start of the function, since that's what'll happen anyway!

Reply:
Thanks @mhl, I didn't know about this!

So does that mean the function becomes this?

```javascript
    $.ajax(endpointUrl, settings).done(function(data) {
      var html, seat_count, seat_counts;

      if (data.results.bindings.length == 1) {
        seat_count = data.results.bindings[0][data.head.vars[0]].value;
        html = '<span class="seatcount">' + seat_count + '</span> seats';
      } else if (data.results.bindings.length > 1) {
        seat_counts = data.results.bindings.map(function(b) {
          return b[data.head.vars[0]].value;
        });
        html = data.results.bindings.length +
                   ' seat count values in Wikidata: <span class="seatcount">' +
                   seat_counts.join('</span>, <span class="seatcount">') +
                   '</span>';
      } else {
        html = 'Number of seats <span class="seatcount">unknown</span> in Wikidata item';
      };

      $('p.seatcount').html(html);
    });
```

```diff
diff --git a/views/house_wikidata.erb b/views/house_wikidata.erb
index e5337946..6460eeab 100644
--- a/views/house_wikidata.erb
+++ b/views/house_wikidata.erb
@@ -32,20 +32,23 @@
         };
 
     $.ajax(endpointUrl, settings).done(function(data) {
+      var html, seat_count, seat_counts;
+
       if (data.results.bindings.length == 1) {
-        var seat_count = data.results.bindings[0][data.head.vars[0]].value;
-        var html = '<span class="seatcount">' + seat_count + '</span> seats';
+        seat_count = data.results.bindings[0][data.head.vars[0]].value;
+        html = '<span class="seatcount">' + seat_count + '</span> seats';
       } else if (data.results.bindings.length > 1) {
-        var seat_counts = data.results.bindings.map(function(b) {
+        seat_counts = data.results.bindings.map(function(b) {
           return b[data.head.vars[0]].value;
         });
-        var html = data.results.bindings.length +
+        html = data.results.bindings.length +
                    ' seat count values in Wikidata: <span class="seatcount">' +
                    seat_counts.join('</span>, <span class="seatcount">') +
                    '</span>';
       } else {
-        var html = 'Number of seats <span class="seatcount">unknown</span> in Wikidata item';
+        html = 'Number of seats <span class="seatcount">unknown</span> in Wikidata item';
       };
+
       $('p.seatcount').html(html);
     });
   });
```

If :+1: I'll make the change and merge.

Reply:
Yes, that's right :+1: